app/models/transaction.rb

```diff
@@ -5,4 +5,8 @@ class Transaction < ApplicationRecord
   def expired?
     created_at < 5.minutes.ago
   end
+
+  def time_until_expiration
+    (created_at + 5.minutes) - Time.current
```

Comment:
Do not use fixed time here.
Add below option in the course.
1. A tutor can set validity for a course purchase (for example, 5 minutes, 1 month, 1 year)